src/lib/input/_input-theme.scss

```diff
@@ -45,5 +45,11 @@
   // be misaligned w.r.t. the placeholder. Adding this margin corrects it.
   input.mat-input-element {
     margin-top: -$line-spacing * 1em;
+    height: $line-height * 1em;
+  }
+
+  input.mat-input-safari {
+    // Hack for 1px input content clipping on Safari:
+    padding-bottom: 1px;
```

Comment:
I was actually thinking we could do the `::first-line` thing here since the `padding` messes up the baseline alignment

Reply:
You're right. I've updated the PR with the `::first-line` approach.